console/console-format-specifier-symbol-manual.html

```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Console Format Specifiers on Symbols</title>
+<meta name="author" title="Dominic Farolino" href="mailto:domfarolino@gmail.com">
+<meta name="assert" content="Console format specifiers on Symbols">
+<link rel="help" href="https://console.spec.whatwg.org/#formatter">
+</head>
+<body>
+<p>Open the console inside the developer tools. It should contain four lines whose contents are:</p>
+<p><code>Symbol(description)</code></p>
+<p><code>Symbol(description)</code></p>
+<p><code>Symbol(description)</code></p>
+
+<script>
+console.log("%s", Symbol.for("description"));
+console.dir("%s", Symbol.for("description"));
```

Comment:
Having `console.dir` use a format string defeats the purpose of `console.dir`.

Reply:
Good point, https://console.spec.whatwg.org/#dir doesn't go through Formatter.

Reply:
Good catch

Reply:
To elaborate, developers use `console.dir(value)` to force a Complete Object Tree representation of whatever `value` was. For example in some tools (Safari) a small Object/Array will show entirely on one line and not be expandable:

![screen shot 2018-01-12 at 11 02 04 am](https://user-images.githubusercontent.com/11351/34890775-136a7940-f788-11e7-9bbc-50e582b33976.png)

But if a developer wants to force this to be interactable, like a large object, they can:

![screen shot 2018-01-12 at 11 02 11 am](https://user-images.githubusercontent.com/11351/34890778-150a4bea-f788-11e7-8836-6326100fbcf0.png)

Another example is logging a Node value. If you `console.log(node)` tools normally give a DOM Tree display of the node, which can be expanded to see children.

![screen shot 2018-01-12 at 11 00 38 am](https://user-images.githubusercontent.com/11351/34890717-de7a846e-f787-11e7-87e9-0079629f73ce.png)

But using `console.dir(node)` tools will give an Object Tree display of the value, and you can expand to see properties and the prototype chain.

![screen shot 2018-01-12 at 11 00 07 am](https://user-images.githubusercontent.com/11351/34890739-f33eadda-f787-11e7-82c3-d4291555a1ca.png)

---

In summary, I don't think this is the direction we want to go for `console.dir`. Its primary purpose is defeated by format strings.

Using a format string as the single value that `console.dir` inspects should produce an Object Tree display of the string value. Likewise, if tools decided to evaluate the format string in `console.dir` it would still just always produce a String.

Reply:
Indeed, I agree this makes sense. That was an error in the tests as I thought (incorrectly from memory) that `console.dir` actually did call the `Formatter` in the spec. Thanks for the good explanation!